contracts/registry_tracker/RegistryTracker.sol

```diff
@@ -0,0 +1,219 @@
+    /// @param _serviceRegistry Service registry address.
+    /// @param _rewardPeriod Reward period in seconds.
+    constructor(address _serviceRegistry, uint256 _rewardPeriod) {
+        // Check for zero addresses
+        if (_serviceRegistry == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Check for zero values
+        if (_rewardPeriod == 0) {
+            revert ZeroValue();
+        }
+
+        serviceRegistry = _serviceRegistry;
+        rewardPeriod = _rewardPeriod;
+
+        owner = msg.sender;
+    }
+
+    /// @dev Changes contract owner address.
+    /// @param newOwner Address of a new owner.
+    function changeOwner(address newOwner) external {
+        // Check for the ownership
+        if (msg.sender != owner) {
+            revert UnauthorizedAccount(msg.sender);
+        }
+
+        // Check for zero address
+        if (newOwner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        owner = newOwner;
+        emit OwnerUpdated(newOwner);
+    }
+
+    /// @dev Changes reward period.
+    /// @param newRewardPeriod New reward period value.
+    function changeRewardPeriod(uint256 newRewardPeriod) external {
+        // Check for the ownership
+        if (msg.sender != owner) {
+            revert UnauthorizedAccount(msg.sender);
+        }
+
+        // Check for zero value
+        if (newRewardPeriod == 0) {
+            revert ZeroAddress();
+        }
+
+        rewardPeriod = newRewardPeriod;
+        emit RewardPeriodUpdated(newRewardPeriod);
+    }
+
+    /// @dev Registers service multisig for registration rewards.
+    /// @param serviceId Service Id.
+    /// @param stakingInstance Staking instance address.
+    function registerMultisig(uint256 serviceId, address stakingInstance) external {
+        // Reentrancy guard
+        if (_locked == 2) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        // Get service multisig and state
+        (, address multisig, , , , , IServiceRegistry.ServiceState serviceState) =
+            IServiceRegistry(serviceRegistry).mapServices(serviceId);
+
+        // Check for multisig address
+        if (multisig == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Check for service state
+        if (serviceState != IServiceRegistry.ServiceState.Deployed) {
+            revert WrongServiceState(serviceId, serviceState);
+        }
+
+        // Check for sender access
+        if (msg.sender != multisig) {
+            // Get service owner
+            address serviceOwner = IServiceRegistry(serviceRegistry).ownerOf(serviceId);
+
+            // Check for service owner
+            if (msg.sender != serviceOwner) {
+                revert UnauthorizedAccount(msg.sender);
+            }
+        }
+
+        // Get service staking state
+        IStaking.StakingState stakingState = IStaking(stakingInstance).getStakingState(serviceId);
```

Comment:
Do we need to be concerned on what exact staking contracts are used? Just thinking out loud ..

Reply:
Staking contracts can be different, but the registration can be done just once. Maybe we need to check something specific about activity checker? Because staking contracts are pretty much all the same.